lua/lean/ft.lua

```diff
@@ -0,0 +1,29 @@
+local M = {}
+
+local find_project_root = require('lspconfig.util').root_pattern('leanpkg.toml')
+
+-- Ideally this obviously would use a TOML parser but yeah choosing to
+-- do nasty things and not add the dependency for now.
+local _MARKER = '.*lean_version.*\".*:3.*'
+
+function M.init3()
```

Comment:
I was trying to keep most of the Lean3 related things all together and in one file -- it's still far off, but that will make it easier to eventually hopefully just delete one file and only need one or two other lines of changes to get rid of it.

Can we put everything here except probably the `.detect()` function itself back in the lean3 file?